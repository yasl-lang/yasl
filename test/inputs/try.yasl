const input = "o:'Draw two'"

const fn index(obj, start, end) -> obj[start:end]
const replace = mt.lookup('', 'replace')

echo try(replace, 'ababsa', 'a', 'A', 2)
echo try(fn() ->  1, 2, 3)
echo try(fn() ->  1, 2)
echo try(fn() ->  1)
echo try((fn(a, b) -> a + b), 9, 10)
const fn f(...) {
    return collections.list(...)
}

echo try(error, '1')
echo try(error, '2')
echo try(error, '3')
echo try(error, 1)
echo try(fn() -> 1, 2, 3)
echo try((fn(a) -> error(a)), '4')


echo try(fn() -> error())
echo try(index, input, 2, 0)
echo try(error, 'sdasda')
echo try(fn() -> ())
echo try(index, input, 0, 2)
