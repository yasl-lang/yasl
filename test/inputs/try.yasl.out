true, AbAbsa, 2
true, 1, 2, 3
true, 1, 2
true, 1
true, 19
false, Error: 1
false, Error: 2
false, Error: 3
false, TypeError: error expected arg in position 0 to be of type str, got arg of type int.
true, 1, 2, 3
false, Error: 4
false, Error
false, ValueError: slicing expected the end of the range to be greater than or equal to the start, got 2:0
false, Error: sdasda
true
true, o:
